packages/jest-matchers/src/matchers.js

```diff
@@ -67,6 +67,18 @@ const matchers: MatchersObject = {
           `Received:\n` +
           `  ${printReceived(received)}`
       : () => {
+          if (
+            getType(received) === getType(expected) &&
+            (getType(received) === 'object' || getType(expected) === 'array') &&
+            equals(received, expected, [iterableEquality])
+          ) {
+            return (
+              matcherHint('.toBe') +
+              '\n\n' +
+              'Looks like you wanted to test for object/array equity with strict `toBe` matcher. You probably need to use `toEqual` instead.'
```

Comment:
Yeah, I think that is preferable. Can you keep it on the same line as the "no visual difference" sentence?